File: Vanilla/sources/zomboid/zombie/config/StringConfigOption.java
package zombie.config;

public class StringConfigOption extends ConfigOption {
   protected String value;
   protected String defaultValue;

   public StringConfigOption(String var1, String var2) {
      super(var1);
      if (var2 == null) {
         var2 = "";
      }

      this.value = var2;
      this.defaultValue = var2;
   }

   public String getType() {
      return "string";
   }

   public void resetToDefault() {
      this.value = this.defaultValue;
   }

   public void setDefaultToCurrentValue() {
      this.defaultValue = this.value;
   }

   public void parse(String var1) {
      this.setValueFromObject(var1);
   }

   public String getValueAsString() {
      return this.value;
   }

   public String getValueAsLuaString() {
      return String.format("\"%s\"", this.value.replace("\\", "\\\\").replace("\"", "\\\""));
   }

   public void setValueFromObject(Object var1) {
      if (var1 == null) {
         this.value = "";
      } else if (var1 instanceof String) {
         this.value = (String)var1;
      } else {
         this.value = var1.toString();
      }

   }

   public Object getValueAsObject() {
      return this.value;
   }

   public boolean isValidString(String var1) {
      return true;
   }

   public void setValue(String var1) {
      if (var1 == null) {
         var1 = "";
      }

      this.value = var1;
   }

   public String getValue() {
      return this.value;
   }

   public String getDefaultValue() {
      return this.defaultValue;
   }
}

File: Vanilla/sources/zdoc/media/lua/shared/Library/StringConfigOption.lua
---@class StringConfigOption : zombie.config.StringConfigOption
---@field protected value String
---@field protected defaultValue String
StringConfigOption = {}

---@public
---@param arg0 String
---@return boolean
function StringConfigOption:isValidString(arg0) end

---@public
---@return String
function StringConfigOption:getValueAsString() end

---@public
---@param arg0 String
---@return void
function StringConfigOption:setValue(arg0) end

---@public
---@param arg0 String
---@return void
function StringConfigOption:parse(arg0) end

---@public
---@return void
function StringConfigOption:resetToDefault() end

---@public
---@return String
function StringConfigOption:getValue() end

---@public
---@param arg0 Object
---@return void
function StringConfigOption:setValueFromObject(arg0) end

---@public
---@return String
function StringConfigOption:getType() end

---@public
---@return void
function StringConfigOption:setDefaultToCurrentValue() end

---@public
---@return Object
function StringConfigOption:getValueAsObject() end

---@public
---@return String
function StringConfigOption:getDefaultValue() end

---@public
---@return String
function StringConfigOption:getValueAsLuaString() end
